slurm/install/install.py

```diff
@@ -257,11 +287,28 @@ def accounting(s: InstallSettings) -> None:
 
 def complete_install(s: InstallSettings) -> None:
     if s.mode == "scheduler":
-        _complete_install_primary(s)
-    _complete_install_all(s)
+        if s.is_primary_scheduler:
+            _complete_install_primary(s)
+        _complete_install_all(s)
+    else:
+        _complete_install_all(s)
 
 
 def _complete_install_primary(s: InstallSettings) -> None:
+    """
+    Only the primary scheduler should be creating files under /sched.
+    """
+    assert s.is_primary_scheduler
+    secondary_scheduler = None
+    if s.secondary_scheduler_name:
+        secondary_scheduler = ilib.await_node_hostname(
```

Comment:
I see the value in this function-- when we move to wanting slurmdbd in its own node (people may prefer that) this would be very useful. But strictly speaking for backup controller we dont need to do this, backup controller host just needs to be up and running but does not have to be before rest of slurm installation takes place. I just think its going to cause delays in converging the cluster. Maybe this can be done as a final check when everything else has been done, and before we start slurmctld?

Reply:
Yes, we definitely need to handle a true topology to be created but for this version we are simply supporting HA alone. People who require HA + an external slurmdbd will have to manage that after converge for this release.